nixos/tests/lxd-qemu.nix

```diff
@@ -0,0 +1,119 @@
+# This tests LXD with useQemu enabled.
+# For now, it tests the same functionality, just with the modified package
+# to ensure nothing broke.
+# Eventually, we will test for VM capabilities.
+
+import ./make-test-python.nix ({ pkgs, lib, ... } :
+
+let
+  lxd-image = import ../release.nix {
+    configuration = {
+      # Building documentation makes the test unnecessarily take a longer time:
+      documentation.enable = lib.mkForce false;
+
+      # Our tests require `grep` & friends:
+      environment.systemPackages = with pkgs; [ busybox ];
+    };
+  };
+
+  lxd-image-metadata = lxd-image.lxdMeta.${pkgs.stdenv.hostPlatform.system};
+  lxd-image-rootfs = lxd-image.lxdImage.${pkgs.stdenv.hostPlatform.system};
+
+in {
+  name = "lxd";
+
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ patryk27 ifd3f ];
+  };
+
+  nodes.machine = { pkgs, lib, ... }: {
+    virtualisation = {
+      diskSize = 4096;
+
+      # Since we're testing `limits.cpu`, we've gotta have a known number of
+      # cores to lean on
+      cores = 2;
+
+      # Ditto, for `limits.memory`
+      memorySize = 512;
+
+      lxc.lxcfs.enable = true;
+      lxd.enable = true;
+    };
+  };
+
+  testScript = ''
+    machine.wait_for_unit("sockets.target")
+    machine.wait_for_unit("lxd.service")
+    machine.wait_for_file("/var/lib/lxd/unix.socket")
+
+    # It takes additional second for lxd to settle
+    machine.sleep(1)
+
+    # lxd expects the pool's directory to already exist
+    machine.succeed("mkdir /var/lxd-pool")
+
+    machine.succeed(
+        "cat ${./common/lxd/config.yaml} | lxd init --preseed"
+    )
+
+    machine.succeed(
+        "lxc image import ${lxd-image-metadata}/*/*.tar.xz ${lxd-image-rootfs}/*/*.tar.xz --alias nixos"
```

Comment:
this is currently running the test using the regular (cgroup) container type

for this to be fixed there would need to be a qemu lxd image. this would be mostly a basic qemu image, plus the lxd defaults and wrapped in tar with metadata